sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,853 @@
+
+    max_constraints : int, optional, default 10 million
+        Maximum number of constraints to enforce per iteration.
+
+    use_sparse : bool, optional, default True
+        Whether to use a sparse matrix (default) or a dense matrix for the
+        impostor-pairs storage. Using a sparse matrix, the distance to
+        impostors is computed twice, but it is somewhat faster for larger
+        data sets than using a dense matrix. With a dense matrix, the unique
+        impostor pairs have to be identified explicitly.
+
+    warm_start : bool, optional, default False
+        When set to True, reuse the solution of the previous
+        call to fit as initialization, otherwise, just erase the
+        previous solution.
+
+    max_corrections : int, optional, default 100
+        The maximum number of variable metric corrections
+        used to define the limited memory matrix. (The limited memory BFGS
+        method does not store the full hessian but uses this many terms in an
+        approximation to it.)
+
+    verbose : bool, optional, default False
+        Whether to print progress messages to stdout.
+
+    iprint : bool, optional, default False
+        Whether to print progress messages from the optimizer.
+
+    random_state : int or numpy.RandomState, optional, default None
+        A seed for reproducibility of random state.
+
+    n_jobs : int, optional, default 1
+        The number of parallel jobs to run for neighbors search.
+        If ``-1``, then the number of jobs is set to the number of CPU cores.
+        Doesn't affect :meth:`fit` method.
+
+    Attributes
+    ----------
+    L_ : array, shape (n_features_out, n_features_in).
+        The linear transformation used during fitting.
+
+    n_neighbors_ : int
+        The provided n_neighbors is decreased when >= min(number of
+        elements in each class).
+
+    n_features_out_ : int
+        The dimensionality of a sample's vector after applying to it the linear
+        transformation.
+
+    classes_ : array-like, shape (n_classes,)
+        The appearing class labels.
+
+    n_iter_ : int
+        The number of iterations of the optimizer.
+
+    n_funcalls_ : int
+        The number of times the optimizer computes the loss and the gradient.
+
+    details_ : dict
+        A dictionary of information created by the L-BFGS optimizer during
+        fitting.
+
+
+    Examples
+    --------
+    >>> X = [[0], [1], [2], [3]]
+    >>> y = [0, 0, 1, 1]
+    >>> from sklearn.neighbors import LargeMarginNearestNeighbor
+    >>> lmnn = LargeMarginNearestNeighbor(n_neighbors=1)
+    >>> lmnn.fit(X, y) # doctest: +ELLIPSIS
+    LargeMarginNearestNeighbor(...)
+    >>> print(lmnn.predict([[1.1]]))
+    [0]
+    >>> print(lmnn.predict_proba([[0.9]]))
+    [[ 1.  0.]]
+
+
+    References
+    ----------
+    .. [1] `Weinberger, K. Q.; Saul L. K. (2009). "Distance Metric Learning
+    for Large Margin Classification". Journal of Machine Learning Research.
```

Comment:
I think that you need to indent for the reference to be well displayed. I cannot check the artifacts right now.